### PR TITLE
Conformance suites: compare actual results CSV with expected results CSV

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
Simple pass/fail results from conformance suite runs aren't always enough information to detect unexpected changes. Comparing all output codes between separate runs can highlight changes that might otherwise be missed.

#### Description of change
Generate `conf-*-actual.csv` on conformance suite runs.
If an expected file exists for the config in `tests/resources/conformance_suites_expected`, generate a diff in HTML format for review.

_Note: the diff is only generated when the conformance suite is run in it's entirety, so shard runs that only run a subset of the conformance suite will not have a diff generated._

#### Steps to Test
I ran the EFM conformance suite with EFM 6.07.31 disabled to demonstrate the HTML diff generation: https://github.com/Arelle/Arelle/actions/runs/8525983010
Example HTML output (GitHub doesn't support HTML file upload here): 
[conf-efm_current-diff.html.txt](https://github.com/Arelle/Arelle/files/14842287/conf-efm_current-diff.html.txt)

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tests/integration_tests/validation/validation_util.py [hunk 1]
```diff
@@ -378,6 +379,42 @@ def load_timing_file(name: str) -> dict[str, float]:
         }
 
 
+def save_actual_results_file(config: ConformanceSuiteConfig, results: list[ParameterSet], diff: bool = True) -> None:
```
Comment (austinmatherne-wk) [inline tests/integration_tests/validation/validation_util.py:382]: nit: this function is doing more than the name implies (it's also maybe creating and saving a diff).

File: tests/integration_tests/validation/run_conformance_suites.py [hunk 2]
```diff
@@ -162,6 +165,7 @@ def run_conformance_suites(
                 series=series_option,
             )
             if log_to_file:
```
Comment (austinmatherne-wk) [inline tests/integration_tests/validation/run_conformance_suites.py:167]: Is `log_to_file` set to `True` in CI?

Comment (aaroncameron-wk) [reply tests/integration_tests/validation/run_conformance_suites.py:167]: Yes, passed [here](https://github.com/Arelle/Arelle/blob/8e907c899c8f79f3607ab8d3a92a4ae7f919cd0d/.github/actions/test_conformance_suite/action.yml#L42).

File: tests/integration_tests/validation/validation_util.py (hunk 1 above)
Comment (aaroncameron-wk) [reply tests/integration_tests/validation/validation_util.py:382]: Good point, will update.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**